src/lib/components/tablev2/SearchWithQueryParams.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { useLocation, useHistory, useRouteMatch } from 'react-router-dom';
+import Search, { SearchProps } from './Search';
+
+export type SearchWithQueryParamsProps = {
+  queryParams?: string,
+} & SearchProps;
+
+export default function SearchWithQueryParams(props) {
+  const { queryParams = 'search', onChange, ...rest } = props;
+  const match = useRouteMatch();
+  const { search } = useLocation();
+  const history = useHistory();
+  const params = new URLSearchParams(search);
+  const initialValue = params.get(queryParams);
+  const [value, setValue] = React.useState(initialValue);
```

Comment:
nit: In order to benefit from tree shaking of React dependency (possible in React 17+) I usually prefer to import only useState and not the global React object